contracts/reflect/src/contract.rs

```diff
@@ -37,13 +38,19 @@ pub fn try_reflect<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
     env: Env,
     msgs: Vec<CosmosMsg<CustomMsg>>,
-) -> StdResult<HandleResponse<CustomMsg>> {
+) -> Result<HandleResponse<CustomMsg>, ReflectError> {
     let state = config(&mut deps.storage).load()?;
-    if deps.api.canonical_address(&env.message.sender)? != state.owner {
-        return Err(StdError::unauthorized());
+
+    let sender = deps.api.canonical_address(&env.message.sender)?;
+    if sender != state.owner {
+        return Err(ReflectError::NotCurrentOwner {
```

Comment:
This is good as it is.

I am curious how this would work with the nightly backtraces. I'd love to see a spike PR adding backtrace support with thiserror to one contract. It probably would need a bit more construction, but good to try out. Also a demo for implementing StdError with thiserror

Reply:
Good point. This can easily be added the same way as in https://github.com/CosmWasm/cosmwasm/pull/537